### PR TITLE
Task: vv/78 add white routes for our public routes

### PR DESCRIPTION
**Pull request by vitalik-prog**

### CODE REVIEW

File: backend/src/common/constants/routes/routes-white-list.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
...
-const ROUTES_WHITE_LIST = [
-  `${ApiPath.AUTH}${AuthApiPath.SIGN_IN}`,
-  `${ApiPath.AUTH}${AuthApiPath.SIGN_UP}`,
-];
+const ROUTES_WHITE_LIST = new Map([
+  [`${ApiPath.AUTH}${AuthApiPath.SIGN_IN}`, [HttpMethod.GET, HttpMethod.POST]],
+  [`${ApiPath.AUTH}${AuthApiPath.SIGN_UP}`, [HttpMethod.GET, HttpMethod.POST]],
+  [`${ApiPath.PODCASTS}${PodcastsApiPath.ROOT}`, [HttpMethod.GET]],
+  [`${ApiPath.EPISODES}${EpisodesApiPath.ROOT}`, [HttpMethod.GET]],
+]);
```
Comment (ghost) [inline backend/src/common/constants/routes/routes-white-list.ts]: use arrays of objects instead

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by Denidzio — DISMISSED** [review]

File: backend/src/middlewares/authorization/authorization.middleware.ts [hunk 2]
```diff
@@ -1,14 +1,19 @@
 import { Response, Request, NextFunction } from 'express';
 import { jwt as jwtMiddleWare } from '../jwt/jwt.middleware';
 
-const authorization = (routesWhiteList: string[] = []) => (
+type WhiteRoute = {
```
Comment (ghost) [inline backend/src/middlewares/authorization/authorization.middleware.ts]: move it to `backend/src/common/types`

File: backend/src/common/constants/routes/routes-white-list.ts [hunk 3]
```diff
@@ -1,8 +1,22 @@
-import { AuthApiPath, ApiPath } from '~/common/enums/enums';
+import { AuthApiPath, ApiPath, PodcastsApiPath, EpisodesApiPath, HttpMethod } from '~/common/enums/enums';
 
 const ROUTES_WHITE_LIST = [
```
Comment (ghost) [inline backend/src/common/constants/routes/routes-white-list.ts]:
```suggestion
const ROUTES_WHITE_LIST: WhiteRoute[] = [
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/middlewares/authorization/authorization.middleware.ts [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
+  path: string;
+  allowedMethods: string[]
+};
+
+const authorization = (routesWhiteList: Array<WhiteRoute> = []) => (
```
Comment (ghost) [inline backend/src/middlewares/authorization/authorization.middleware.ts]:
```suggestion
const authorization = (routesWhiteList: WhiteRoute[] = []) => (
```

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by Sayonarko — DISMISSED** [review]

File: backend/src/middlewares/authorization/authorization.middleware.ts [hunk 5]
```diff
@@ -1,14 +1,19 @@
-import { Response, Request, NextFunction } from 'express';
```
Comment (what1s1ove) [inline backend/src/middlewares/authorization/authorization.middleware.ts:1]: rename the file
`backend/src/middlewares/check-auth/check-auth.middleware.ts`

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Pull request merged**